src/inmanta/db/util.py

```diff
@@ -161,3 +163,72 @@ async def clear_database(postgresql_client: Connection) -> None:
         ",".join(type_names),
         ",".join(function_names),
     )
+
+
+class ColumnDefinition(NamedTuple):
+    """
+    :param name: The name of the column.
+    :param is_list: A boolean that indicates whether this column has the type list.
+    :param default: The default value of this column. Or None, when this column doesn't have a default value.
+    """
+
+    name: str
+    is_list: bool = False
+    default: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class EnumUpdateDefinition:
+    """
+    A definition on how an existing enum in the database has to be updated.
+
+    :param name: The name of the enumeration.
+    :param values: The values the enum should have after the update.
+    :param deleted_values: A dictionary that indicates which elements are deleted from the existing enum and how
+                           they should be migrated. The key of the dictionary is the name of the removed enum value
+                           and the value of the dictionary is the value it should be replaced with or None if the new value
+                           should be NULL.
+    :param columns: A dictionary that indicates which columns of which tables are using the enum.
+                    The key of the dictionary contains the name of the table.
+    """
+
+    name: str
+    values: abc.Sequence[str]
+    deleted_values: abc.Mapping[str, Optional[str]]  # deleted values mapped to new value if any currently exist
+    columns: abc.Mapping[str, abc.Sequence[ColumnDefinition]]  # columns with defaults
```

Comment:
Your description in the docstring is way better than the comments I added here. I'd drop them.
```suggestion
    deleted_values: abc.Mapping[str, Optional[str]]
    columns: abc.Mapping[str, abc.Sequence[ColumnDefinition]]
```